src/signal_client.py

```
@@ -79,50 +82,44 @@ async def _listen(self):
-                    self.received_signals[signal_type]["received"].append(signal_data)
-
-    # Used to set up how many instances of a signal to wait for, before triggering the actions
-    # that cause them to be emitted.
-    async def prepare_wait_for_signal(self, signal_type: str, delta_count: int, accept_fn: Optional[Callable] = None):
-        if signal_type not in self.await_signals:
+        if signal_type in self.signal_queues:
+            await self.signal_queues[signal_type].put(signal_data)
+            logger.debug(f"Queued signal: {signal_type}")
+        else:
+            logger.debug(f"Ignored signal not in await list: {signal_type}")
+
+    async def wait_for_signal(self, signal_type: str, timeout: int = 20) -> dict:
+        if signal_type not in self.signal_queues:
             raise ValueError(f"Signal type {signal_type} is not in the list of awaited signals")
-        async with self.signal_lock:
-            self.received_signals[signal_type]["delta_count"] = delta_count
-            self.received_signals[signal_type]["expected_count"] = (
-                len(self.received_signals[signal_type]["received"]) + delta_count
-            )
-            self.received_signals[signal_type]["accept_fn"] = accept_fn
-
-    async def wait_for_signal(self, signal_type: str, timeout: int = 20) -> dict | list[dict]:
-        if signal_type not in self.await_signals:
+        try:
+            signal = await asyncio.wait_for(self.signal_queues[signal_type].get(), timeout)
+            logger.info(f"Received {signal_type} signal: {signal}")
+            return signal
+        except asyncio.TimeoutError:
+            raise TimeoutError(f"Signal {signal_type} not received in {timeout} seconds")
+
+    async def signal_stream(self, signal_type: str) -> AsyncGenerator[dict, None]:
+        if signal_type not in self.signal_queues:
             raise ValueError(f"Signal type {signal_type} is not in the list of awaited signals")
```

I think maybe we should just return instead of raising ValueError here.

If we do something like this:
```
async for value in client.signal_stream(some_signal):
    ...
```
I think the expected behavior would be that nothing happens. Just like if we did:
```
for value in []:
    print(value)
```
We expect nothing to print.

---

Hm I think it is better to raise an error because we are creating defined signal_queues for the node. To me it doesn't make sense to do a `get` in a list that you are not subscribed. It is probably a mistake by the person that is calling the function.